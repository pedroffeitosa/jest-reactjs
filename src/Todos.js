import React from 'react';

import { connect } from 'react-redux';

function Todos({ todos }) {
  return (
    <ul className="Todos">
      {todos.map(todo=>(
        <li key={todo.id}>{todo.text}</li>
      ))}
    </ul>
  );
}

const mapStateToProps = state => ({
  todos: state.todos
})

export default connect(mapStateToProps)(Todos);
