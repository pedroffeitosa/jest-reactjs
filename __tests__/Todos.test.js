import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Todos from '../src/Todos';

const mockStore = configureStore();

const initialState = {
    todos: [
      { id: 0, text: "Fazer CafÃ©"},
      { id: 1, text: "Acessar o slack"},
      { id: 2, text: "Ver RocketLive"}
  ]
}

const store = mockStore(initialState);

describe('Testing Todos Component',()=>{
  it('should render correctly', ()=> {
    const wrapper = shallow(   
      <Todos store={store}/>
      ).dive()
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  })
})