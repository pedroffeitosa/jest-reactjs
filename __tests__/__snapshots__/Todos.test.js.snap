// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Todos Component should render correctly 1`] = `
<Todos
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  todos={
    Array [
      Object {
        "id": 0,
        "text": "Fazer CafÃ©",
      },
      Object {
        "id": 1,
        "text": "Acessar o slack",
      },
      Object {
        "id": 2,
        "text": "Ver RocketLive",
      },
    ]
  }
/>
`;
